ubg/src/mafia/MafiaNight.js

```diff
@@ -30,44 +30,101 @@ const useStyles = makeStyles((theme) => ({
 export default function MafiaNight({user, usersData, room,
   mafiaKill, doctorSave, detectiveCheck}) {
   const classes = useStyles();
+  const [initialize, setInitialize] = React.useState(false);
   const [players, setPlayers] = React.useState([]);
   const [userInfo, setUserInfo] = React.useState('');
   const [roleText, setRoleText] = React.useState('');
+  const [message, setMessage] = React.useState('');
 
   /**
    * @return {undefined}
    */
   function loadNightData() {
-    const allPlayers = [];
-    usersData.forEach(function(u) {
-      if (u.alive === true) {
-        allPlayers.push(u);
-      }
-      if (u.uid === user.uid) {
-        setUserInfo(u);
-        if ( u.role=== 1) {
-          setRoleText('Pretend to be clicking or thinking :)');
-        } else if (u.role === 2) {
-          setRoleText('Mafia, pick someone to kill.');
-        } else if (u.role === 3) {
-          setRoleText('Detective, who do you want to check tonight?');
-        } else if (u.role === 4) {
-          setRoleText('Doctor, who do you want to save tonight?');
+    if (mafiaKill['uid'] !== '' &&
```

Comment:
I'm just curious, why are you using mafiaKill['uid'], instead of mafiaKill.uid? I mean both work just fine.